File: src/test/java/tictactoe/CompletedGamesAnalysisTest.java
package tictactoe;

import org.junit.jupiter.api.Test;
import tictactoe.CompletedGamesReport;

import java.io.ByteArrayInputStream;
import java.io.InputStream;
import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static tictactoe.CompletedGamesAnalysis.generateCompletedGamesReport;

public class CompletedGamesAnalysisTest {

    @Test
    public void canGenerateReport() {
        List<String> lines = Arrays.asList("XX=OOO===", "XX=OOO===", "XX=OOO===", "XOX=OX=O=", "XX=OOO===", "XOXOXOXOX", "=========");
        CompletedGamesReport result = generateCompletedGamesReport(new ByteArrayInputStream(String.join("\n", lines).getBytes()));
        assertEquals(
                "Winner Summary \n" +
                        "==============\n" +
                        "Player X: 1\n" +
                        "Player O: 5\n" +
                        "Draw: 1\n",
                result.getContent()
        );
    }

    @Test
    public void canGenerateReportForHugeFile() {
        InputStream inputStream = CompletedGamesAnalysisTest.class.getResourceAsStream("/huge_completed_games.txt");
        CompletedGamesReport result = generateCompletedGamesReport(inputStream);
        assertEquals(
                "Winner Summary \n" +
                        "==============\n" +
                        "Player X: 1267200\n" +
                        "Player O: 1224960\n" +
                        "Draw: 464640\n",
                result.getContent()
        );
    }
}

File: src/test/kotlin/ticktactoe/CompletedGamesAnalysisTest.kt
package ticktactoe

import tictactoe.CompletedGamesReport
import tictactoe.generateCompletedGamesReport
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class CompletedGamesAnalysisTest {

    @Test
    fun `can generate report`(){
        val lines = listOf("XX=OOO===", "XX=OOO===", "XX=OOO===", "XOX=OX=O=", "XX=OOO===", "XOXOXOXOX", "=========")
        val result = generateCompletedGamesReport(lines.joinToString("\n").byteInputStream())
        assertEquals(CompletedGamesReport(1, 5, 1), result)
        assertEquals("""
            Winner Summary 
            ==============
            Player X: 1 
            Player O: 5 
            Draw: 1 
        """.trimIndent(), result.getContent())
    }

    @Test
    fun `can generate report for huge file`(){
        val inputStream = CompletedGamesAnalysisTest::class.java.getResourceAsStream("/huge_completed_games.txt")
        val result = generateCompletedGamesReport(inputStream)
        assertEquals(CompletedGamesReport(1267200, 1224960, 464640), result)
        assertEquals("""
            Winner Summary 
            ==============
            Player X: 1267200 
            Player O: 1224960 
            Draw: 464640 
        """.trimIndent(), result.getContent())
    }
}